src/App.svelte

```diff
@@ -44,12 +64,34 @@
 		alert(`Don't give up! You lost 1 life!`)
 		hero.life-=1;
 	}
+
+	function buyItem(event){
+		const newItem = event.detail.product
+		const productPrice = newItem.price
+		let updatedItems = [...items, newItem]
+		hero.gold -= productPrice
```

Comment:
Please use 'newItem.price' instead of 'productPrice' to avoid unnecessary use of memory.